eng/build.yml

```diff
@@ -177,3 +179,138 @@ jobs:
         publishLocation: Container	
       continueOnError: true	
       condition: always()
+
+- ${{ if in(parameters.agentOs, 'Windows_NT', 'Darwin') }}:
```

Comment:
FYI @marcpopMSFT since this PR is adding new build infrastructure to the SDK.

Reply:
There is definitely some risk adding additional legs/tests as it could reduce our overall reliability which is already somewhat problematic. How confident are we that these tests are going to pass? Will the tests work with the arcade test retry functionality (that hasn't been enabled yet but we want to)?  Them running in parallel alleviates perf concerns I might have but the reliability is the main one not (and just the complexity of maintaining the build.yml)

Reply:
I was going to bring this up in an email, but looks like the conversation has started already.

This was a sensitive area for us as well, @marcpopMSFT, so we approached it very carefully, knowing that it can be a source of concern. The fact that @TanayParikh did some runs and built up the chart in the description is to evaluate both performance as well as reliability. All the runs that he kicked off passed successfully and you can see that there has been 9 runs in that data set and not just a few.
On top of that, we plan to monitor this leg very closely after merging for a week or so to make sure we catch anything that may come up and react quickly.

Having said that, I think we should get this merged now while @TanayParikh is around so that he can monitor this in the next few weeks.